packages/lib/ren/src/index.ts

```diff
@@ -145,13 +146,17 @@ export default class RenJS {
             new SimpleLogger((config && config.logLevel) || LogLevel.Error);
 
         this._config.logger = this._logger;
-
-        // Use provided provider, provider URL or default lightnode URL.
-        this.renVM =
-            providerOrNetwork &&
-            typeof providerOrNetwork !== "string" &&
-            (providerOrNetwork as AbstractRenVMProvider).sendMessage
-                ? (providerOrNetwork as AbstractRenVMProvider)
+        const defaultProvider =
```

Comment:
It might make sense to add a `() =>` in front of this (and then call `defaultProvider()` below) so that the providers don't get created if the user has provided their own.

Reply:
Will do

Reply:
Fixed this